### PR TITLE
Fix uploading of include and exclude from multi string array #398

### PR DESCRIPTION
**Pull request by Bracciata**

### CODE REVIEW

**Comment by michael-siek** [comment]
Thank you very much for your contribution!

**Review by Zidious — CHANGES_REQUESTED** [review]
Hey @Bracciata! 

Thanks for contributing, unfortunately the issue I created was not quite correct. The current include / exclude selectors I put in the issue [ticket](https://github.com/dequelabs/axe-core-npm/issues/398) show iframe selectors and not the correct layout for include/excluding whom parent include said classes as shown in in the axe-core example:

> Include the whole document except any structures whose parent contains the class `exclude1` or `exclude2`. the structure would need to look like: 

```js
const builder = new AxeBuilder()
  .exclude([ ['.exclude1'], ['.exclude2'] ])
  ```

File: packages/playwright/tests/axe-playwright.spec.ts [hunk 1]
```diff
@@ -459,48 +459,38 @@ describe('@axe-core/playwright', () => {
...
-        '.selector-three'
-      ]);
+      const results = await new AxeBuilder({ page })
+        .include(['.include', '.include2'])
+        .analyze();
 
-      const { include: actual } = (await results.analyze()) as any;
+      const flattenTarget = flatPassesTargets(results);
 
-      assert.deepEqual(actual[0], expected);
```
Comment (Zidious) [inline packages/playwright/tests/axe-playwright.spec.ts:480]: We would need to keep this test, it checks that the selectors are getting passed to axe-core

File: packages/playwright/tests/axe-playwright.spec.ts [hunk 2]
```diff
@@ -459,48 +459,38 @@ describe('@axe-core/playwright', () => {
...
+    });
+    it('with chaining include/exclude using an array of strings', async () => {
+      await page.goto(`${addr}/context.html`);
+      const results = await new AxeBuilder({ page })
+        .include(['.include', '.include2'])
+        .exclude(['.exclude', '.exclude2'])
+        .analyze();
+      const flattenTarget = flatPassesTargets(results);
 
-      assert.deepEqual(actual[0], expected);
```
Comment (Zidious) [inline packages/playwright/tests/axe-playwright.spec.ts:503]: We would need to keep this exclude test, it checks that the exclude selectors are getting passed to axe-core

**Comment by Bracciata** [comment]
@Zidious What are your thoughts on fully flattening the array then appending?

**Comment by Zidious** [comment]
> @Zidious What are your thoughts on fully flattening the array then appending?

@Bracciata Thanks for getting back to me! I believe that will not work as its how it gets passed to axe-core. Currently, we allow chaining of include/exclude which produces the correct array format, example: 

```js
const builder = new AxeBuilder()
  .exclude('.exclude1')
  .exclude('.exclude2')
  
  // When produces this when passed to axe-core: { exclude: [ [ '.exclude' ], [ '.exclude2' ] ] }
```
So we would need to mimic the same behaviour, leave it with me, I will need to discuss it further with the team and I will get back to you ASAP.

**Comment by Zidious** [comment]
@Bracciata Sorry to keep you waiting, had a discussion with the team due to how the include / exclude array is passed to axe-core it would require a significant amount of changes. We have decided to keep the current include / exclude chaining for classes (See below): 

```js
const builder = new AxeBuilder()
  .exclude('.exclude1')
  .exclude('.exclude2')
  ```

**Pull request closed without merging**